cmd/ttn-lw-stack/commands/as_db.go

```diff
@@ -227,6 +227,41 @@ var (
 			return nil
 		},
 	}
+	asDBUplinkRegistryCleanupCommand = &cobra.Command{
+		Use:   "cleanup-uplink-registry",
+		Short: "Clean stale Application Server uplink registry data",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			logger.Info("Connecting to Redis database...")
+			cl := ttnredis.New(config.Redis.WithNamespace("as", "applicationups"))
+
+			logger.Info("Cleaning uplink registry data")
+
+			numCleared := 0
+			uidApplicationUpsKey := cl.Key("uid", "*")
+			pipeliner := cl.Pipeline()
+			err := ttnredis.RangeRedisKeys(ctx, cl, uidApplicationUpsKey, ttnredis.DefaultRangeCount,
+				func(k string) (bool, error) {
+					if err := pipeliner.Del(ctx, k).Err(); err != nil {
```

Comment:
A pipeline is executed only when you call `Exec` - individual calls to the pipeline cannot fail.